fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/AdminShowReplicaStatusCommand.java

```diff
@@ -0,0 +1,154 @@
+import org.apache.doris.nereids.analyzer.UnboundSlot;
+import org.apache.doris.nereids.trees.expressions.EqualTo;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.Not;
+import org.apache.doris.nereids.trees.expressions.literal.StringLikeLiteral;
+import org.apache.doris.nereids.trees.plans.PlanType;
+import org.apache.doris.nereids.trees.plans.commands.info.TableRefInfo;
+import org.apache.doris.nereids.trees.plans.visitor.PlanVisitor;
+import org.apache.doris.qe.ConnectContext;
+import org.apache.doris.qe.ShowResultSet;
+import org.apache.doris.qe.ShowResultSetMetaData;
+import org.apache.doris.qe.StmtExecutor;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * ShowReplicaStatusCommand
+ */
+public class AdminShowReplicaStatusCommand extends ShowCommand {
+    public static final ImmutableList<String> TITLE_NAMES = new ImmutableList.Builder<String>()
+            .add("TabletId").add("ReplicaId").add("BackendId").add("Version").add("LastFailedVersion")
+            .add("LastSuccessVersion").add("CommittedVersion").add("SchemaHash").add("VersionNum")
+            .add("IsBad").add("IsUserDrop").add("State").add("Status")
+            .build();
+
+    private TableRefInfo tableRefInfo;
+    private Expression where;
+
+    private Replica.ReplicaStatus statusFilter;
+
+    public AdminShowReplicaStatusCommand(TableRefInfo tableRefInfo, Expression where) {
+        super(PlanType.SHOW_REPLICA_STATUS_COMMAND);
+        this.tableRefInfo = tableRefInfo;
+        this.where = where;
+    }
+
+    @Override
+    public ShowResultSet doRun(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        validate(ctx);
+        List<List<String>> results;
+        try {
+            results = MetadataViewer.getTabletStatus(tableRefInfo.getTableNameInfo().getDb(),
+                tableRefInfo.getTableNameInfo().getTbl(),
+                tableRefInfo.getPartitionNamesInfo().getPartitionNames(), statusFilter, where);
+        } catch (DdlException e) {
+            throw new AnalysisException(e.getMessage());
+        }
+        return new ShowResultSet(getMetaData(), results);
+    }
+
+    public ShowResultSetMetaData getMetaData() {
+        ShowResultSetMetaData.Builder builder = ShowResultSetMetaData.builder();
+        for (String title : TITLE_NAMES) {
+            builder.addColumn(new Column(title, ScalarType.createVarchar(30)));
+        }
+        return builder.build();
+    }
+
+    /**
+     * validate for show replica
+     */
+    public void validate(ConnectContext ctx) throws AnalysisException, UserException {
+        // check auth
+        if (!Env.getCurrentEnv().getAccessManager().checkGlobalPriv(ConnectContext.get(), PrivPredicate.ADMIN)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "ADMIN");
+        }
+
+        // bind relation
+        tableRefInfo.analyze(ctx);
+
```

Comment:
is Util.prohibitExternalCatalog(tblRef.getName().getCtl(), this.getClass().getSimpleName()); missed?

Reply:
done